### PR TITLE
Add color functions and giant color comment

### PR DESCRIPTION
**Pull request by keithburgie**
### Do not merge. 

@errolgh, check out `theme.ts` below.

I made a function where you put in some colors:
```typescript
const semanticPalette: Record<SemanticKey, HexColor> = {
  primary: "#e63946", // crimson
  secondary: "#457B9D", // steelBlue
  background: "#f1faee", // mint
  info: "#A8DADC", // cyan
  dark: "#1D3557", // navyBlue
};

const semanticColors = createSemanticColors(semanticPalette);

export default semanticColors;
```
Then in `tailwind.config.js` you import those colors:
```javascript
import semanticColors from "./theme";

/** @type {import('tailwindcss').Config}*/
export default {
  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
  theme: {
    extend: {
      colors: semanticColors,
    },
  },
  plugins: [],
};
```
Then in `Button.tsx` you make Tailwind your ~~bitch~~ lil' croissant:
```jsx
return (
    <>
      <button className={`${accessibleHtmlButton} mb-4`} {...props}>
        Button
      </button>

      <div className="grid grid-cols-11">
        {semanticColorKeys.map((color) =>
          shadeRange.map((shade) => (
            <button key={color + shade} className={`bg-${color}-${shade}`}>
              {shade}
            </button>
          ))
        )}
      </div>
    </>
  );
```
Screenshot 2023-10-02 at 9 05 47 PM

## THE STUPIDEST PART
Tailwind decides which CSS to include by scanning the files for actual complete strings. That's why their docs include this warning:
Screenshot 2023-10-02 at 9 14 25 PM

I've bypassed this by creating `color-comment.ts`, which is just a giant comment of all the possible colors generated from the config file. This file isn't imported anywhere, it just exists. And because it exists, **you can use these colors**.

TBD if it will still work when the application is "built" for production. We'll see!

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,10 +24,14 @@
...
   "types": "./lib/index.d.ts",
   "dependencies": {
+    "@types/chroma-js": "^2.4.1",
     "autoprefixer": "^10.4.16",
+    "chroma-js": "^2.4.2",
+    "classnames": "^2.3.2",
     "postcss": "^8.4.31",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "tailwind-merge": "^1.14.0",
```
Comment (keithburgie) [inline package.json:34]: Remove or figure out how this can be more helpful.

File: .eslintrc.cjs
Comment (keithburgie) [inline .eslintrc.cjs:1]: This just got auto formatted.

File: src/components/Button/Button.stories.tsx
Comment (keithburgie) [inline src/components/Button/Button.stories.tsx:1]: WIP

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -1,12 +1,83 @@
-import React from "react";
-import { Button as ElButton } from "@spectral-labs/elements-react";
+import React, { ButtonHTMLAttributes } from "react";
+import classnames from "classnames";
+import { twMerge } from "tailwind-merge";
 
-interface ButtonProps {
-  label: string;
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  // colorScheme: string;
```
Comment (keithburgie) [inline src/components/Button/Button.tsx:6]: WIP

File: tailwind.config.js [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+import semanticColors from "./theme";
+
 /** @type {import('tailwindcss').Config}*/
 export default {
   content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
   theme: {
-    extend: {},
+    extend: {
+      // colors: semanticColors,
+      colors: semanticColors,
```
Comment (keithburgie) [inline tailwind.config.js:11]: Probably time for a break.

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -1,12 +1,77 @@
-import React from "react";
-import { Button as ElButton } from "@spectral-labs/elements-react";
```
Comment (errolgh) [inline src/components/Button/Button.tsx:2]: I got an error here? It broke storybook. This line fixed it on my end. We should containerize. Looking up AWS EKS prices rn 🏃🏾‍♂️.

fix:
`import { Button as ElButton } from "../Button/index"`

**Pull request closed without merging**